code/modules/mob/emote.dm

```diff
@@ -1,132 +1,89 @@
+	else if(copytext(input,1,6) == "yells")
+		to_chat(user, "<span class='danger'>Invalid emote.</span>")
+	else if(copytext(input,1,5) == "asks")
+		to_chat(user, "<span class='danger'>Invalid emote.</span>")
 	else
-		input = message
-	if(input)
-		message = "<B>[src]</B> [input]"
-	else
-		return
-
-
-	if(message)
-		log_emote(message, src)
-
- //Hearing gasp and such every five seconds is not good emotes were not global for a reason.
- // Maybe some people are okay with that.
-
-		for(var/mob/M in player_list)
-			if(!M.client)
-				continue //skip monkeys and leavers
-			if(istype(M, /mob/new_player))
-				continue
-			if(findtext(message," snores.")) //Because we have so many sleeping people.
-				break
-			if(M.stat == DEAD && M.get_preference(CHAT_GHOSTSIGHT) && !(M in viewers(src,null)))
-				M.show_message(message)
-
-
-		// Type 1 (Visual) emotes are sent to anyone in view of the item
-		if(m_type & 1)
-			var/list/can_see = get_mobs_in_view(1,src)  //Allows silicon & mmi mobs carried around to see the emotes of the person carrying them around.
-			can_see |= viewers(src,null)
-			for(var/mob/O in can_see)
-
-				if(O.status_flags & PASSEMOTES)
-
-					for(var/obj/item/holder/H in O.contents)
-						H.show_message(message, m_type)
-
-					for(var/mob/living/M in O.contents)
-						M.show_message(message, m_type)
-
-				O.show_message(message, m_type)
-
-		// Type 2 (Audible) emotes are sent to anyone in hear range
-		// of the *LOCATION* -- this is important for pAIs to be heard
-		else if(m_type & 2)
-			for(var/mob/O in get_mobs_in_view(7,src))
-
-				if(O.status_flags & PASSEMOTES)
-
-					for(var/obj/item/holder/H in O.contents)
-						H.show_message(message, m_type)
-
-					for(var/mob/living/M in O.contents)
-						M.show_message(message, m_type)
-
-				O.show_message(message, m_type)
-
-/mob/proc/emote_dead(var/message)
-	if(client.prefs.muted & MUTE_DEADCHAT)
-		to_chat(src, "<span class='warning'>You cannot send deadchat emotes (muted).</span>")
-		return
-
-	if(!(client.prefs.toggles & CHAT_DEAD))
-		to_chat(src, "<span class='warning'>You have deadchat muted.</span>")
-		return
-
-	if(!src.client.holder)
-		if(!config.dsay_allowed)
-			to_chat(src, "<span class='warning'>Deadchat is globally muted</span>")
-			return
-
-
-	var/input
-	if(!message)
-		input = sanitize(copytext(input(src, "Choose an emote to display.") as text|null, 1, MAX_MESSAGE_LEN))
-	else
-		input = message
-
-	if(input)
-		message = "<span class='game deadsay'><span class='prefix'>DEAD:</span> <b>[src]</b> [message]</span>"
+		. = FALSE
+
+/datum/emote/mob/custom/run_emote(mob/user, params, type_override = null)
+	if(user.client && user.client.prefs.muted & MUTE_IC)
+		to_chat(user, "You cannot send IC messages (muted).")
+		return FALSE
+	if(user.client && user.client.handle_spam_prevention(params, MUTE_IC))
+		return FALSE
+
+	if(!params)
+		var/custom_emote = copytext(sanitize(input("Choose an emote to display.") as text|null), 1, MAX_MESSAGE_LEN)
```

Comment:
Actually making this legacy code work properly is awful. The only way to do custom emotes should really be the me verb.